app/models/infrastructure_facades/pl_grid_facade.rb

```diff
@@ -70,8 +70,16 @@ def start_simulation_managers(user_id, instances_count, experiment_id, additiona
 
     if additional_params[:onsite_monitoring]
       sm_uuid = SecureRandom.uuid
-      PlGridFacade.send_and_launch_onsite_monitoring(credentials, sm_uuid, user_id,
+      begin
```

Comment:
Rzeczywiście, to najwygodniejsze rozwiązanie. Chociaż kod który podałeś nie zadziała (&block przyjmie podany blok i yield nie będzie miało co wykonać).